jax/lax/lax.py

```diff
@@ -2236,35 +2236,44 @@ def _dot_general_transpose_rhs(g, x, dimension_numbers, precision):
 
 def _dot_general_batch_rule(batched_args, batch_dims, dimension_numbers,
                             precision):
+  # there are three kinds of dimensions in a dot_general:
+  # - contraction dimensions appear in lhs and rhs but not the result
+  # - batch dimensions appear in lhs, rhs, and result
+  # - other dimensions appear in the result and one of lhs or rhs
```

Comment:
Maybe we could call that a tensor product dimension? Same below for "adding an other dimension".